hack/skipci/main.go

```diff
@@ -0,0 +1,118 @@
+package main
+
+/*
+
+A list of available environment variables can be found here:
+
+https://github.com/kubernetes/test-infra/blob/a9f967cb1235916fb5a4eca661fa083413211242/prow/jobs.md#job-environment-variables
+
+For periodics:
+- JOB_NAME
+- JOB_TYPE
+- JOB_SPEC
+- BUILD_ID
+- PROW_JOB_ID
+
+*/
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+)
+
+var (
+	commit = flag.String("commit", "HEAD~1..HEAD", "commit to compare against. defaults to n-1")
+	debug  = flag.Bool("debug", false, "whether to print debug statements")
+)
+
+type commitFile struct {
+	extension string
+	directory string
+	filename  string
+	original  string
+}
+
+var (
+	whiteListDirectories = []string{
+		"docs/",
+		"hack/skipci",
+	}
+
+	whiteListFileExt = []string{
+		".md",
+		".asciidoc",
+	}
+)
+
+func whiteListed(f commitFile, debug bool) bool {
+	for _, wld := range whiteListDirectories {
+		if strings.Contains(f.directory, wld) {
+			if debug {
+				fmt.Printf("matched dir (%s) on whitelist dir (%s) with (%s)\n", f.original, f.directory, wld)
+			}
+			return true
+		}
+	}
+	for _, wle := range whiteListFileExt {
+		if strings.HasSuffix(f.extension, wle) {
+			if debug {
+				fmt.Printf("matched file (%s) on whitelist file extension (%s) with (%s)\n", f.original, f.extension, wle)
+			}
+			return true
+		}
+	}
+
+	return false
+}
+
+func getCommitFiles(commit string) ([]commitFile, error) {
+	cmd := "git"
+	args := []string{"diff-tree", "--no-commit-id", "--name-only", "-r", commit}
```

Comment:
i think this should be something like the follow to be useful for multicommit PRs
```bash
git diff-tree --no-commit-id --name-only -r HEAD..origin/master
```